api/src/main/java/jakarta/persistence/criteria/NullPrecedence.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2008, 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Gavin King      - 3.2
+
+package jakarta.persistence.criteria;
+
+/**
+ * Specifies the precedence of null values within query result sets.
+ *
+ * @see CriteriaBuilder#asc(Expression, NullPrecedence)
+ * @see CriteriaBuilder#desc(Expression, NullPrecedence)
+ *
+ * @since 3.2
+ */
+public enum NullPrecedence {
```

Comment:
I'm _almost_ inclined to just call this enum `Nulls`. WDYT?

Reply:
let's go with `Nulls`